### PR TITLE
cartservice - fix proto file path to build independently

### PR DESCRIPTION
**Pull request by cijothomas**
Change the path to the proto file in the csproj, without which we cannot build/run the cartservice independently.

### CODE REVIEW

File: src/cartservice/src/cartservice.csproj [hunk 1]
```diff
@@ -17,6 +17,6 @@
   </ItemGroup>
 
   <ItemGroup>
-    <Protobuf Include="**\*.proto" GrpcServices="Both" />
+    <Protobuf Include="..\..\..\pb\*.proto" GrpcServices="Both" />
```
Comment (julianocosta89) [inline src/cartservice/src/cartservice.csproj:20]: I've pulled this PR and tested locally, but whenever trying to build the cartservice again, I got the following issue:

```shell
 => ERROR [ghcr.io/open-telemetry/demo:0.1.0-cartservice builder 6/6] RUN dotnet publish ./src/cartservice.csproj -p:PublishSingleFile=true -r linux-  2.9s
------
 > [ghcr.io/open-telemetry/demo:0.1.0-paymentservice internal] load build context:
------
------
 > [ghcr.io/open-telemetry/demo:0.1.0-cartservice builder 6/6] RUN dotnet publish ./src/cartservice.csproj -p:PublishSingleFile=true -r linux-musl-x64 --self-contained true -p:PublishTrimmed=True -p:TrimMode=Link -c Release -o /cartservice --no-restore:
#0 0.737 Microsoft (R) Build Engine version 17.2.0+41abc5629 for .NET
#0 0.737 Copyright (C) Microsoft Corporation. All rights reserved.
#0 0.737
#0 2.813 /usr/src/app/src/services/CartService.cs(19,7): error CS0246: The type or namespace name 'Hipstershop' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.813 /usr/src/app/src/services/CartService.cs(23,32): error CS0246: The type or namespace name 'Hipstershop' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.813 /usr/src/app/src/cartstore/ICartStore.cs(26,14): error CS0246: The type or namespace name 'Hipstershop' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.813 /usr/src/app/src/cartstore/LocalCartStore.cs(74,21): error CS0246: The type or namespace name 'Hipstershop' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.813 /usr/src/app/src/services/CartService.cs(33,51): error CS0246: The type or namespace name 'AddItemRequest' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.813 /usr/src/app/src/services/CartService.cs(33,36): error CS0246: The type or namespace name 'Empty' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.813 /usr/src/app/src/services/CartService.cs(44,44): error CS0246: The type or namespace name 'GetCartRequest' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.813 /usr/src/app/src/services/CartService.cs(44,30): error CS0246: The type or namespace name 'Cart' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.813 /usr/src/app/src/services/CartService.cs(53,53): error CS0246: The type or namespace name 'EmptyCartRequest' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.813 /usr/src/app/src/services/CartService.cs(53,36): error CS0246: The type or namespace name 'Empty' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.813 /usr/src/app/src/cartstore/RedisCartStore.cs(176,27): error CS0246: The type or namespace name 'Hipstershop' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.814 /usr/src/app/src/services/CartService.cs(25,33): error CS0246: The type or namespace name 'Empty' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.814 /usr/src/app/src/cartstore/LocalCartStore.cs(26,46): error CS0246: The type or namespace name 'Hipstershop' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
#0 2.814 /usr/src/app/src/cartstore/LocalCartStore.cs(27,26): error CS0246: The type or namespace name 'Hipstershop' could not be found (are you missing a using directive or an assembly reference?) [/usr/src/app/src/cartservice.csproj]
------
failed to solve: executor failed running [/bin/sh -c dotnet publish ./src/cartservice.csproj -p:PublishSingleFile=true -r linux-musl-x64 --self-contained true -p:PublishTrimmed=True -p:TrimMode=Link -c Release -o /cartservice --no-restore]: exit code: 1
```

**Review by julianocosta89 — CHANGES_REQUESTED** [review]

File: src/cartservice/src/cartservice.csproj (hunk 1 above)
Comment (cijothomas) [reply src/cartservice/src/cartservice.csproj:20]: ok i need to try something more then. thanks for reviewing!

**Comment by mic-max** [comment]
Do you mean independently from other services? That can be achieved with `docker compose build cartservice [--no-cache]` and run with `docker compose up cartservice`

Or do you mean independently as-in without using docker compose? Which I know would not be possible without copying the `.proto` file to the correct directory or changing the path in the `.csproj` as shown in this PR. I think it's fine if we rely on docker compose, since it is also bringing in stuff from the `.env` file. I'm not sure if this will remain the case when we add k8s, anyone who knows more on this please share I'd like to learn!

**Comment by cijothomas** [comment]
> Do you mean independently from other services? That can be achieved with `docker compose build cartservice [--no-cache]` and run with `docker compose up cartservice`
> 
> Or do you mean independently as-in without using docker compose? Which I know would not be possible without copying the `.proto` file to the correct directory or changing the path in the `.csproj` as shown in this PR. I think it's fine if we rely on docker compose, since it is also bringing in stuff from the `.env` file. I'm not sure if this will remain the case when we add k8s, anyone who knows more on this please share I'd like to learn!

I was referring to dotnet build ...csproj , which obviously failed without this PR.

**Comment by mic-max** [comment]
> > Do you mean independently from other services? That can be achieved with `docker compose build cartservice [--no-cache]` and run with `docker compose up cartservice`
> > Or do you mean independently as-in without using docker compose? Which I know would not be possible without copying the `.proto` file to the correct directory or changing the path in the `.csproj` as shown in this PR. I think it's fine if we rely on docker compose, since it is also bringing in stuff from the `.env` file. I'm not sure if this will remain the case when we add k8s, anyone who knows more on this please share I'd like to learn!
> 
> I was referring to dotnet build ...csproj , which obviously failed without this PR.

Why not use `docker compose build`? Changing the `.csproj` file will also require changing the service's `Dockerfile`

**Comment by cartersocha** [comment]
@cijothomas are we closing this pr or are you going to come back to it?

**Comment by cijothomas** [comment]
> @cijothomas are we closing this pr or are you going to come back to it?

Please close. I think Michaels suggestion makes sense, so I dont need this anymore.

**Pull request closed without merging**